lib/pure/collections/sequtils.nim

```diff
@@ -325,6 +565,94 @@ proc map*[T, S](s: openArray[T], op: proc (x: T): S {.closure.}):
   for i in 0 ..< s.len:
     result[i] = op(s[i])
 
+proc mapIndexed*[T, S](s: openArray[T], op:
+                       proc (i: int, x: T): S {.closure.}): seq[S]{.inline.} =
+  ## Returns a new sequence with the results of `op` proc applied to every
+  ## item together with its index in the container `s`.
+  ##
+  ## Since the input is not modified you can use it to
+  ## transform the type of the elements in the input container.
+  ##
+  ## See also:
+  ## * `mapIt template<#mapIt.t,typed,untyped>`_
+  ## * `apply proc<#apply,openArray[T],proc(T)_2>`_ for the in-place version
+  ##
+  runnableExamples:
+    let
+      a = @[1, 2, 3, 4]
+      b = mapIndexed(a, proc(i: int, x: int): int = x - i)
+    assert b == @[1, 1, 1, 1]
+
+  newSeq(result, s.len)
+  for i, v in s:
+    result[i] = op(i, s[i])
+
+proc scan*[T](s: openArray[T], op: proc (x: T) {.closure.})
+                                                {.since: (1, 1).} =
```

Comment:
Nit: is this really the best way to format this code? I would suggest:

```nim
proc scan*[T](
  s: openArray[T], op: proc (x: T) {.closure.}
) {.since: (1, 1).} =
```

Reply:
I would suggest increasing the column limit to 120 :P